### ISSUE TITLE
paypal bundle ID

### ISSUE THREAD

**Issue by jamesdixon** [Taracque/cordova-plugin-braintree#1]
Hi @Taracque,

First off, thanks again for all of your hard work on this plugin. I was finally able to get it to build locally for me.

However, as mentioned in the other repo, I'm still seeing the following error when choosing PayPal as my payment method.

```
[BraintreeSDK] CRITICAL PayPal requires [BTAppSwitch setReturnURLScheme:] to be configured to begin with your app's bundle ID (com.scoutforpets.customer). Currently, it is set to (com.scoutforpets.customer.payments).
```

Is there something that needs to be configured in my application to make this work?

Thanks!

**Comment by Taracque** [comment]
Please check what is in your platform/ios/PROJECTNAME-Info.plist file.

It should be the following:
```
<key>CFBundleURLTypes</key>
<array>
    <dict>
        <key>CFBundleTypeRole</key>
        <string>Editor</string>
        <key>CFBundleURLSchemes</key>
        <array>
            <string>${PRODUCT_BUNDLE_IDENTIFIER}.payments</string>
        </array>
    </dict>
</array>
```

**Comment by jamesdixon** [comment]
@Taracque it turns out that block was missing from my plist. Looking at the `plugin.xml` for the io-dropin branch, it doesn't appear to be added.

That said, adding the above code didn't work until I manually replaced `${PRODUCT_BUNDLE_IDENTIFIER}` with my application id. Is that a variable that needs to be created somewhere?

**Comment by Taracque** [comment]
The above should be added by the https://github.com/Taracque/cordova-plugin-braintree/blob/master/hooks/add_embedded_ios_frameworks.js hook. Which is an after_plugin_add hook. The PRODUCT_BUNDLE_IDENTIFIER is a variable in your project.pbxproj file, this variable is automatically replaced by the Product Bundle Identifier defined in Target Build Settings in XCode.

**Comment by jamesdixon** [comment]
> The above should be added by the https://github.com/Taracque/cordova-plugin-braintree/blob/master/hooks/add_embedded_ios_frameworks.js hook. Which is an after_plugin_add hook.

I don't see anything referencing that in `master`, but I do see it in `io-dropin`. I'm not sure why it's not being added to my plist given that I'm using that branch.

> The PRODUCT_BUNDLE_IDENTIFIER is a variable in your project.pbxproj file, this variable is automatically replaced by the Product Bundle Identifier defined in Target Build Settings in XCode.

I did find the variable you are referencing. Interestingly, it was empty even though my `config.xml` contains `<widget id="com.scoutforpets.customer"...`.

**Comment by championboylay** [comment]
Same to me. I also still facing this issue.

**Comment by Taracque** [comment]
Please check what is in your platform/ios/PROJECTNAME-Info.plist file.
see: https://github.com/Taracque/cordova-plugin-braintree/issues/1#issuecomment-271993731

**Comment by championboylay** [comment]
Firstly thanks to Taracque. I added as you suggested [#1 (comment)](https://github.com/Taracque/cordova-plugin-braintree/issues/1#issuecomment-271993731). But it didn't still soved.

I found [Add URLs to Whitelist](https://developers.braintreepayments.com/guides/paypal/client-side/ios/v4). I added the following to my application's info.plist. It is worked now.

`<key>LSApplicationQueriesSchemes</key>`
`<array>`
 `  <string>com.paypal.ppclient.touch.v1</string>`
  ` <string>com.paypal.ppclient.touch.v2</string>`
`   <string>com.venmo.touch.v2</string>`
`</array>`

**Comment by ionut-tanasa** [comment]
I think this issue is now fixed by #14 and [d9155f1](https://github.com/Taracque/cordova-plugin-braintree/commit/d9155f1310147d7e9c9662703dbacd43b38d7a2c)
